source/opt/instruction.h

```diff
@@ -86,6 +86,7 @@ class Instruction {
               std::vector<Instruction>&& dbg_line = {});
 
   SpvOp opcode() const { return opcode_; }
+  void SetOpcode(SpvOp opcode) { opcode_ = opcode; }
```

Comment:
This is potentially unsafe.  This method should get a blurb, and the blurb should mention that in many cases operands will have to change!